WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved

```diff
@@ -64,6 +64,15 @@
           "version": "0.2.3"
         }
       },
+      {
+        "package": "SwiftSoup",
+        "repositoryURL": "https://github.com/scinfu/SwiftSoup.git",
```

Comment:
The size of the dependency is a bit concerning:

```
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Swift                           58           1709           3707          10315
```

It should not affect incremental builds, I think it's something to be considered.

Reply:
I would suggest considering wrapping https://developer.wordpress.org/block-editor/reference-guides/packages/packages-blocks/ and other related WordPress JS APIs in Objective-C and invoking them using [JavaScriptCore](https://developer.apple.com/documentation/javascriptcore). This way, we'll be able to reuse the existing parser. You [can run](https://developer.apple.com/documentation/javascriptcore/jsvirtualmachine#Support-Threading-and-Concurrent-JavaScript-Execution) Javascript in background on iOS to ensure that processing doesn't take a lot of time.

A native package that would wrap the existing JS packages could be useful for more scenarios than this. For example, the app could use it for addressing the "dangling" media uploads issue.

Reply:
> The size of the dependency is a bit concerning:
> [...]
> It should not affect incremental builds, I think it's something to be considered.

I haven't checked how much this library increases the binary. If it's significant we might consider other libraries.

> I would suggest considering wrapping https://developer.wordpress.org/block-editor/reference-guides/packages/packages-blocks/ and other related WordPress JS APIs in Objective-C and invoking them using [JavaScriptCore](https://developer.apple.com/documentation/javascriptcore). This way, we'll be able to reuse the existing parser. You [can run](https://developer.apple.com/documentation/javascriptcore/jsvirtualmachine#Support-Threading-and-Concurrent-JavaScript-Execution) Javascript in background on iOS to ensure that processing doesn't take a lot of time.
> 
> A native package that would wrap the existing JS packages could be useful for more scenarios than this. For example, the app could use it for addressing the "dangling" media uploads issue.

This approach is intriguing and would certainly ensure consistent results when processing a block compared to Gutenberg. As a long-term solution, I'd advocate exploring this approach. However, I believe it would entail significant complexity in implementation, particularly regarding how to bundle the necessary code and potential tweaks to the Gutenberg code. @kean not sure if with your comment you're proposing following this instead of the refactor implemented in this PR. WDYT?

Reply:
> your comment you're proposing following this instead of the refactor implemented in this PR. WDYT?

It's more long-term. I worked with JS-based packages before, so I know it should be feasible, but I don't know how much effort it would take.

> particularly regarding how to bundle the necessary code

These packages are already part of the Gutenberg-mobile project, aren't they?

Reply:
> These packages are already part of the Gutenberg-mobile project, aren't they?

Yes, but we bundled them with [Metro](https://metrobundler.dev/) and files are compiled with [Hermes](https://github.com/facebook/hermes). I presume the output of this setup won't be supported natively. We'd need to generate a new bundle compatible with `JavaScriptCore`.

Reply:
The app doesn't need to use these JS files directly. It could part of the existing Gutenberg framework.